rest_framework/fields.py

```diff
@@ -690,6 +690,12 @@ class BooleanField(Field):
     }
     NULL_VALUES = {'null', 'Null', 'NULL', '', None}
 
+    def __init__(self, **kwargs):
+        if kwargs.get('allow_null', False):
```

Comment:
allow null false or true? can you elaborate please

Reply:
HTML checkboxes do not send any value and should be treated
        as `None` by BooleanField if allow_null is True.  from a comment of the tests in this PR

Reply:
If allow_null is true (false if not set) then set those default values, that was the problem,. If not set, It was consideres as false on booleanfield, It needs another widget to have the there values (none, true, false). As nullboolean field addesed that but It has been deprecated, the current booleanfield treats unexistant as false, the fix correcta that as show in the tests.

Reply:
I think for compatibility reason we can consider this PR. just have to carefully handle any breaking or compatibility breaking changes. may be we can just mention that on release notes later.